apm-agent-plugins/apm-log-correlation-plugin/src/main/java/co/elastic/apm/agent/mdc/MdcActivationListener.java

```diff
@@ -180,8 +180,13 @@ public void beforeActivate(ErrorCapture error) throws Throwable {
 
     public void before(TraceContext traceContext, boolean isError) throws Throwable {
         if (loggingConfiguration.isLogCorrelationEnabled() && tracer.isRunning()) {
+            ClassLoader applicationClassLoader = traceContext.getApplicationClassLoader();
             for (WeakKeySoftValueLoadingCache<ClassLoader, MethodHandle> mdcPutMethodHandleCache : mdcPutMethodHandleCaches) {
-                MethodHandle put = mdcPutMethodHandleCache.get(getApplicationClassLoader(traceContext));
```

Comment:
As long as you properly remove, adding to any available MDC is conceptually correct because it just stores thread-context data. The related logging system, whatever it may be, can either use or ignore this info, which is correct in the context of this thread.

Reply:
Ok, I will adapt the PR then.